rig/scripts/rig_boot.py

```diff
@@ -0,0 +1,114 @@
+                        help="width of SpiNNaker system")
+    parser.add_argument("height", type=int, default=None, nargs="?",
+                        help="height of SpiNNaker system")
+    parser.add_argument("--hardware-version", type=int, default=None,
+                        help="board hardware version number")
+    parser.add_argument("--led-config", type=int, default=None,
+                        help="LED configuration word")
+
+    parser.add_argument("--binary", type=str, default=None,
+                        help="binary to boot the system with")
+
+    # Automatically build a list of available machine parameters by inspecting
+    # boot module.
+    type_group = parser.add_mutually_exclusive_group()
+    for dict_name in dir(boot):
+        if dict_name.endswith(BOOT_OPTION_POSTFIX):
+            type_name = dict_name[:-len(BOOT_OPTION_POSTFIX)]
+            option_name = "--{}".format(type_name)
+            option_dict = getattr(boot, dict_name)
+            option_help = "use predefined boot options for a {} board".format(
+                type_name)
+            type_group.add_argument(option_name, action="store_const",
+                                    const=option_dict, default={},
+                                    dest="machine_type",
+                                    help=option_help)
+
+    args = parser.parse_args(args)
+
+    # Check that either a width and height have been supplied or a predefined
+    # set of machine parameters was chosen.
+    if (args.width is None and args.height is None and
+            args.machine_type == {}):
+        parser.error("either width and height must be specified "
+                     "or a predefined boot option selected.")
+    elif (args.width is None) ^ (args.height is None):
+        parser.error("if used, both width and height must be specified")
+
+    # Accumulate the set of options from the commandline
+    options = {
+        "hardware_version": 0,
+        "led_config": 0x00000001,
+        "boot_data": (open(args.binary, "rb").read()
+                      if args.binary is not None else None),
+    }
+    options.update(args.machine_type)
+    options.update({
+        opt_name: getattr(args, opt_name)
+        for opt_name in "width height hardware_version led_config".split()
+        if getattr(args, opt_name) is not None
+    })
+
+    # See if the device is already booted
+    mc = MachineController(args.hostname, n_tries=1)
+    try:
+        info = mc.get_software_version(0, 0)
+        if "SpiNNaker" in info.version_string:
+            sys.stderr.write(
+                "{}: error: system already booted with {} v{}\n".format(
+                    parser.prog,
+                    info.version_string.split("/")[0],
+                    info.version))
+            return 1
+        else:
+            sys.stderr.write(
+                "{}: error: host is not a SpiNNaker system "
+                "({} running {} v{})\n".format(
```

Comment:
Correct. I've done this and been confused when what follows is an apparently successful boot.